### PR TITLE
fix: Added support for django42

### PR DESCRIPTION
**Pull request by zubairshakoorarbisoft**
[Issue](https://github.com/openedx/edx-analytics-data-api/issues/664)

### CODE REVIEW

File: analytics_data_api/v0/views/courses.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def get_csv_filename(self):
         return f'{course_id}--{self.slug}.csv'
 
     def finalize_response(self, request, response, *args, **kwargs):
-        if request.META.get('HTTP_ACCEPT') == 'text/csv':
+        if request.headers.get('accept') == 'text/csv':
```
Comment (awais786) [inline analytics_data_api/v0/views/courses.py:66]: Revert this change

File: analytics_data_api/v0/models.py [hunk 2]
```diff
@@ -135,7 +135,7 @@ class Meta(BaseCourseModel.Meta):
 
     module_id = models.CharField(db_index=True, max_length=255)
     part_id = models.CharField(db_index=True, max_length=255)
-    correct = models.NullBooleanField()
+    correct = models.BooleanField(null=True)
```
Comment (UsamaSadiq) [inline analytics_data_api/v0/models.py:138]: Run `make migrations` in the shell. It'll introduce new migrations due to this change.

**Review by UsamaSadiq — CHANGES_REQUESTED** [review]

File: analytics_data_api/v0/models.py (hunk 2 above)
Comment (iamsobanjaved) [reply analytics_data_api/v0/models.py:138]: Doc link: https://docs.djangoproject.com/en/4.2/releases/4.0/#features-removed-in-4-0

File: analytics_data_api/v0/views/courses.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def get(self, request, *args, **kwargs):
         self.course_id = self.kwargs.get('course_id')
         start_date = request.query_params.get('start_date')
         end_date = request.query_params.get('end_date')
-        timezone = utc
+        timezone = datetime.timezone.utc
```
Comment (iamsobanjaved) [inline analytics_data_api/v0/views/courses.py:35]: #### Reason for this change

> The [django.utils.timezone.utc](https://docs.djangoproject.com/en/4.2/ref/utils/#django.utils.timezone.utc) alias to [datetime.timezone.utc](https://docs.python.org/3/library/datetime.html#datetime.timezone.utc) is deprecated. Use [datetime.timezone.utc](https://docs.python.org/3/library/datetime.html#datetime.timezone.utc) directly.

https://github.com/edx/video-encode-manager/blob/6cf71026b8c5964d980dcd57abf41a46ee1976a6/Makefile#L106

**Review by iamsobanjaved — APPROVED** [review]

**Review by UsamaSadiq — APPROVED** [review]

**Review by Zacharis278 — APPROVED** [review]
